protocol/client/database.go

```diff
@@ -180,6 +182,10 @@ func (db SQLLiteDatabase) SaveContacts(contacts []Contact) (err error) {
 	if err != nil {
 		return err
 	}
+	history, err := tx.Prepare("INSERT INTO history_topics(contact_id, synced) VALUES (?, ?)")
```

Comment:
I wonder if we should use `contact_id` or `topic` as a foreign key. It kinda makes `history_topics` artificially connected to `user_contacts` table while it could be independent and `topic` can be more generic.

EDIT: I noticed that you couple it this way so that history items are deleted on cascade. I don't see any usecase now that could justify `topic` instead of `contact_id`. Unless we really embrace the mental model where `contact` and `topic` are 1-1 related.

If we want to keep this, I would rename the table to `history_user_contact_topic` to emphasize that it's 1-1 related to `user_contacts` table.

Reply:
Also, shouldn't have `OR REPLACE INTO` as well?

Reply:
i imagined that we can have multiple chats listening on same topic. for example even in current implementation i can create public chat with contact-discovery name and it will overlap with every private topic. but this is a separate chat and we will have to request all history for this topic again.

i probably should remove replace in the previous statement because we can't replace it here, it will overwrite synced value

Reply:
That's true. We can't guarantee that topics will be unique and the probability that the same topic will be used for, for instance, a private and public chat will only increase.

> If we want to keep this, I would rename the table to history_user_contact_topic to emphasize that it's 1-1 related to user_contacts table.

I would still consider this.

> i probably should remove replace in the previous statement because we can't replace it here, it will overwrite synced value

+1 `synced` value needs to be carried over.